lib/client/workflow-client/workflowClient.js

```diff
@@ -0,0 +1,367 @@
+var q = require('q');
+var _ = require('lodash');
+var shortid = require('shortid');
+var CONSTANTS = require('../../constants');
+var MediatorTopicUtility = require('fh-wfm-mediator/lib/topics');
+
+/**
+ *
+ * Getting Promises for done and error topics.
+ * This will resolve or reject the returned promise depending on the topic published.
+ *
+ *
+ * TODO: This may be of more use in fh-wfm-mediator...
+ *
+ * @param doneTopicPromise  - A promise for the done topic.
+ * @param errorTopicPromise - A promise for the error topic.
+ * @returns {Promise}
+ */
+function getTopicPromises(doneTopicPromise, errorTopicPromise) {
+  var deferred = q.defer();
+
+  doneTopicPromise.then(function(createdWorkorder) {
+    deferred.resolve(createdWorkorder);
+  });
+
+  errorTopicPromise.then(function(error) {
+    deferred.reject(error);
+  });
+
+  return deferred.promise;
+}
+
+/**
+ *
+ * A mediator service that will publish and subscribe to topics to be able to render workflow data.
+ *
+ * @param {Mediator} mediator
+ * @param {object}   config
+ * @constructor
+ */
+function WorkflowMediatorService(mediator, config) {
+  this.mediator = mediator;
+  this.config = config || {};
+
+  this.resultsTopics = new MediatorTopicUtility(mediator).prefix(CONSTANTS.TOPIC_PREFIX).entity(CONSTANTS.RESULTS_ENTITY_NAME);
+  this.workordersTopics = new MediatorTopicUtility(mediator).prefix(CONSTANTS.TOPIC_PREFIX).entity(CONSTANTS.WORKORDER_ENTITY_NAME);
+  this.usersTopics =  new MediatorTopicUtility(mediator).prefix(CONSTANTS.TOPIC_PREFIX).entity(CONSTANTS.USERS_ENTITY_NAME);
+  this.workflowSyncSubscribers = new MediatorTopicUtility(mediator).prefix(CONSTANTS.SYNC_TOPIC_PREFIX).entity(CONSTANTS.WORKFLOW_ENTITY_NAME);
+}
+
+
+/**
+ *
+ * Getting Promises for the done and error topics.
+ *
+ * TODO: This may be of more use in fh-wfm-mediator...
+ *
+ * @param {MediatorTopicUtility} topicGenerator
+ * @param {string} topicName   - The name of the topic to generate
+ * @param {string} [topicUid]  - A topic UID if required.
+ * @returns {Promise} - A promise for the topic.
+ */
+WorkflowMediatorService.prototype.getErrorAndDoneTopicPromises = function getErrorAndDoneTopicPromises(topicGenerator, topicName, topicUid) {
+  var doneTopic = topicGenerator.getTopic(topicName, CONSTANTS.DONE_PREFIX, topicUid);
```

Comment:
Please consider breaking this into two lines. We should try to avoid the need to scroll right to read full line.